repository/src/main/java/org/apache/atlas/repository/store/graph/v2/preprocessor/sql/QueryFolderPreProcessor.java

```diff
@@ -80,12 +91,211 @@ private void processCreate(AtlasStruct entity) throws AtlasBaseException {
         entity.setAttribute(QUALIFIED_NAME, createQualifiedName(collectionQualifiedName));
     }
 
-    private void processUpdate(AtlasStruct entity, AtlasVertex vertex) {
-        String vertexQnName = vertex.getProperty(QUALIFIED_NAME, String.class);
+    private void processUpdate(AtlasEntity entity, AtlasVertex vertex, EntityMutationContext context) throws AtlasBaseException {
+        AtlasEntityType entityType      = typeRegistry.getEntityTypeByName(entity.getTypeName());
+        AtlasObjectId newParentAttr     = (AtlasObjectId) entity.getRelationshipAttribute(PARENT_ATTRIBUTE_NAME);
+        String relationshipType         = AtlasEntityUtil.getRelationshipType(newParentAttr);
+        AtlasAttribute parentAttribute  = entityType.getRelationshipAttribute(PARENT_ATTRIBUTE_NAME, relationshipType);
+        AtlasObjectId currentParentAttr = (AtlasObjectId) entityRetriever.getEntityAttribute(vertex, parentAttribute);
```

Comment:
Rename currentParentAttr to currentParentObjectId

Reply:
Done